src/compiler/checker.ts

```diff
@@ -26566,7 +26566,10 @@ namespace ts {
 
                 if (computedNameType && !(computedNameType.flags & TypeFlags.StringOrNumberLiteralOrUnique)) {
                     if (isTypeAssignableTo(computedNameType, stringNumberSymbolType)) {
-                        if (isTypeAssignableTo(computedNameType, numberType)) {
+                        if (computedNameType.flags & TypeFlags.Any) {
+                            hasComputedStringProperty = true;

```

Comment:
I think we want `string`, `number`, and `symbol` index signatures, not just string.